src/test/java/com/rultor/agents/AgentsITCase.java

```diff
@@ -0,0 +1,768 @@
+            issue.comments().iterate(),
+            Matchers.<Comment>iterableWithSize(2)
+        );
+        MatcherAssert.assertThat(
+            new Comment.Smart(issue.comments().get(2)).body(),
+            Matchers.containsString(
+                AgentsITCase.PHRASES
+                    .getString("QnIfPull.already-closed")
+            )
+        );
+    }
+
+    /**
+     * Make profile.
+     * @return Profile
+     * @throws IOException If fails
+     */
+    private static Profile profile() throws IOException {
+        final Profile profile = Mockito.mock(Profile.class);
+        Mockito.doReturn(
+            new XMLDocument(
+                String.format(
+                    "<p><entry key='architect'><item>%s</item></entry></p>",
+                    AgentsITCase.ARCHITECT
+                )
+            )
+        ).when(profile).read();
+        Mockito.doReturn(
+            new ArrayMap<String, InputStream>().with(
+                "file.bin",
+                new ByteArrayInputStream(
+                    // @checkstyle MagicNumber (1 line)
+                    new byte[]{0, 1, 7, 8, 9, 10, 13, 20}
+                )
+            )
+        ).when(profile).assets();
+        return profile;
+    }
+
+    /**
+     * Make talk from issue.
+     * @param issue The issue
+     * @return Talk
+     * @throws IOException If fails
+     */
+    private Talk talk(final Issue issue) throws IOException {
+        Assume.assumeFalse(SystemUtils.IS_OS_WINDOWS);
+        final SSHD sshd = new SSHD(this.temp.newFolder());
+        final int port = sshd.port();
+        final String executor;
+        if (SystemUtils.IS_OS_LINUX) {
+            executor = "md5sum";
+        } else {
+            executor = "md5";
+        }
+        final Talk talk = new Talk.InFile();
+        talk.modify(
+            new Directives().xpath("/talk")
+                // @checkstyle MultipleStringLiterals (1 line)
+                .add("shell").attr("id", "abcdef")
+                .add("host").set("localhost").up()
+                .add("port").set(Integer.toString(port)).up()
+                .add("login").set(sshd.login()).up()
+                .add("key").set(sshd.key()).up().up()
+                .add("daemon").attr("id", "fedcba")
+                .add("title").set("some operation").up()
+                .add("script")
+                .set(
+                    String.format("ls -al; %s file.bin; sleep 50000", executor)
+                )
+                .up()
+                .up()
+                .attr("later", "true")
+                .add("wire")
+                .add("href").set("http://test2").up()
+                .add("github-repo").set(issue.repo().coordinates().toString())
+                .up()
+                .add("github-issue").set(Integer.toString(issue.number())).up()
+        );
+        return talk;
+    }
+
+    /**
+     * Mock github.
+     */
+    private static final class MockGithub implements Github {
```

Comment:
@dskalenko why do we need a wrapper for `MkGithub` here ?? What functionality does this add ?
I see a number of pure wrappers here! Please if you need this explain why in the PR description and remove obviously redundant wrappers.